Tests/SwiftFormatPrettyPrintTests/GarbageTextTests.swift

```diff
@@ -0,0 +1,147 @@
+
+final class GarbageTextTests: PrettyPrintTestCase {
+  func testHashBang() {
+    let input =
+      """
+      #!/usr/bin/swift -foo -bar
+      print("Hello world!")
+      """
+
+    let expected =
+      """
+      #!/usr/bin/swift -foo -bar
+      print(
+        "Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 20)
+  }
+
+  func testBOM() {
+    let input =
+      """
+      \(bom)print("Hello world!")
+      """
+
+    let expected =
+      """
+      \(bom)print("Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 21)
+  }
+
+  func testBOMPresenceDoesNotPermitLeadingNewlines() {
+    let input =
+      """
+      \(bom)
+      print("Hello world!")
+      """
+
+    let expected =
+      """
+      \(bom)print("Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 21)
+  }
+
+  func testUnknownCodePointAsLeadingTrivia() {
+    let input =
+      """
+      \(unknownScalar)print("Hello world!")
+      """
+
+    let expected =
+      """
+      \(unknownScalar)print("Hello world!")
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 21)
+  }
+
+  func testUnknownCodePointAsTrailingTriviaAreGluedToPreviousToken() {
+    // Note: The third line here is a bit of a weird case. Despite the operator being separated from
+    // the left-hand-side by a space, the intervening garbage text appears to put the parser in a
+    // mode where it treats the operator as a postfix operator, which then causes `secondTerm` to be
+    // parsed as its own distinct statement and `CodeBlockItem`---hence the lack of indentation.
+    let input =
+      """
+      x = y\(unknownScalar)+z
+
+      x = firstTerm\(unknownScalar)+secondTerm
+
+      x = firstTerm \(unknownScalar)+ secondTerm
+
+      x = firstTerm\(unknownScalar) + secondTerm
+
+      x = firstTerm \(unknownScalar) + secondTerm
+
+      x = firstTerm \(unknownScalar) \(unknownScalar) + secondTerm
+      """
+
+    let expected =
+      """
+      x = y\(unknownScalar) + z
+
+      x =
+        firstTerm\(unknownScalar)
+        + secondTerm
+
+      x = firstTerm \(unknownScalar)+
+      secondTerm
+
+      x =
+        firstTerm\(unknownScalar)
+        + secondTerm
+
+      x =
+        firstTerm \(unknownScalar)
+        + secondTerm
+
+      x =
+        firstTerm \(unknownScalar) \(unknownScalar)
+        + secondTerm
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 20)
+  }
+
+  func testConflictMarkers() {
+    let input =
+      """
+      func greet() {
+      print("Hello")
+      <<<<<<< hash:filename
+      print("werld")
+      =======
+      print("world")
+      >>>>>>> hash:filename
+      print("!!")
+      }
+      """
+
+    let expected =
+      """
+      func greet() {
+        print("Hello")
+      <<<<<<< hash:filename
+      print("werld")
```

Comment:
Should this line and the next non-garbage line be indented to match the scope? Or are these lines also treated as `garbageText`?

Reply:
Everything from the opening conflict marker to the closing marker is lexed as part of the garbage text, so we don't have a way to format what's inside either branch.